### PR TITLE
feat: SITES-23397 Stop a running import job

### PR DESCRIPTION
**Pull request by atopper**
Fix https://jira.corp.adobe.com/browse/SITES-23397

Test URLs:
- Before: https://main--helix-labs-website--adobe.aem.live/tools/import/index.html
- After: https://stop-import--helix-labs-website--adobe.aem.live/tools/import/index.html

Show a "Stop job" button during an import, that stops the job.

### CODE REVIEW

File: tools/import/import.js [hunk 1]
```diff
@@ -177,6 +191,7 @@ function addJobsList(jobs) {
       downloadLink.download = 'import_results.zip';
       resultsContainer.append(downloadLink);
     }
+    displayStopButton(job.status === 'RUNNING');
```
Comment (swetabar) [inline tools/import/import.js:194]: Wondering if for later, we could provide a Stop option regardless of whether or not a job was started from labs-ui?

**Review by swetabar — APPROVED** [review]

File: tools/import/index.html [hunk 2]
```diff
@@ -71,6 +71,10 @@ <h2>Job Details</h2>
                     <div id="results"></div>
                 </div>
             </div>
+            <p class="button-wrapper hidden" id="stop-import-job">
+                <button type="button" class="button">Stop Job</button>
```
Comment (arumsey) [inline tools/import/index.html:75]: I think this should be `Stop job` (lower-case j).

File: tools/import/import.js (hunk 1 above)
Comment (arumsey) [reply tools/import/import.js:194]: If you know the job id I think the labs UI will fetch te results of that job. What we need is an API to fetch all the jobs for an api-key.

File: tools/import/index.html (hunk 2 above)
Comment (arumsey) [reply tools/import/index.html:75]: Also, a default cta button may be too much for a stop action. Should it match the download button styling which adds the `accent` class?

File: tools/import/import.js (hunk 1 above)
Comment (atopper) [reply tools/import/import.js:194]: Maybe add all the api-key's jobs in the dropdown of jobs (in addition to the ones in localstorage)...
Some day.

File: tools/import/index.html (hunk 2 above)
Comment (atopper) [reply tools/import/index.html:75]: How about a "negative" variant?  We don't want them clicking it accidentally (there is no "Are you sure?")

File: tools/import/import.js [hunk 3]
```diff
@@ -288,4 +305,13 @@ function addJobsList(jobs) {
       });
     }
   });
+
+  fields.stopButton.addEventListener('click', async (event) => {
+    event.preventDefault();
+    try {
+      await service.stopCurrentJob();
```
[the remark this answers is not in the document]
Comment (atopper) [reply tools/import/import.js:312]: Hmm...  Reworked that a few times.  All errors are now in the catch, so maybe...  I'll fix it up.

File: tools/import/import.css [hunk 4]
```diff
@@ -108,10 +108,21 @@
...
+.import button.negative,
+.import button.negative:focus {
+  background-color: var(--red-800);
+  color: var(--gray-100);
+}
+
+.import button.negative:hover {
+  color: var(--color-text);
+  border-color: var(--red-800);
+}
```
Comment (arumsey) [inline tools/import/import.css:124]: If you want to go down the path of intriducing a negative variant I think there are more states you need to handle. Also, an outline button may be more appropriate.

https://spectrum.adobe.com/page/button/#Negative-variant

image

image

Comment (arumsey) [reply tools/import/import.css:124]:
```
.import button.negative, .import button.negative:focus {
    background-color: var(--grey-100);
    color: var(--red-800);
    border-color: var(--red-800);
}

.import button.negative:focus {
    outline-color: var(--red-800);
}

.import button.negative:hover {
    background-color: var(--red-100);
    color: var(--red-800);
}

```

Comment (arumsey) [reply tools/import/import.css:124]: image

File: tools/import/import.css [hunk 5]
```diff
@@ -108,10 +108,26 @@
...
 }
 
-.import div.hidden {
+.import .hidden {
   display: none;
 }
 
+.import button.negative,
+.import button.negative:focus {
+  background-color: var(--gray-100);
```
Comment (arumsey) [inline tools/import/import.css:117]: I think this is my final suggestion. Make the background white.

```suggestion
  background-color: var(--color-background);
```

**Review by arumsey — APPROVED** [review]

**Pull request merged**